webviz_subsurface/_private_plugins/tornado_plot.py

```diff
@@ -574,3 +575,27 @@ def calc_low_x(low, high):
         base = min(0, high)
         return low - base
     return 0
+
+
+def printable_int_list(integer_list: Optional[List[int]]):
+    """Creates a string out of a list of integers.
+    The string gives a range x-y if all the integers between and
+    including x and y are in the list, otherwise separated by commas.
+    Example: the list [0, 1, 2, 4, 5, 8, 10] becomes '0-2, 4-5, 8, 10'
+    If the input is `None` or the list is empty, the string 'None' is returned.
+    """
+
+    def _printable_int_list_rec(int_list):
+        if len(int_list) == 1:
+            return f"{int_list[0]}"
+        for i in range(1, len(int_list)):
+            if int_list[i] != (int_list[i - 1] + 1):
+                if i == 1:
+                    return f"{int_list[0]}, {_printable_int_list_rec(int_list[i:])}"
+                return f"{int_list[0]}-{int_list[i-1]}, {_printable_int_list_rec(int_list[i:])}"
+        return f"{int_list[0]}-{int_list[-1]}"
+
+    if integer_list is not None and len(integer_list) > 0:
+        integer_list = sorted(integer_list)
+        return _printable_int_list_rec(integer_list)
+    return "None"
```

Comment:
```suggestion
    if not integer_list:
        return "None"

    sorted_list = sorted(integer_list)
    prev_number = sorted_list[0]
    string = str(prev_number)

    for real in sorted_list[1:]:
        if real > prev_number + 1:
            string += (f"{prev_number}" if string.endswith("-") else "") + f", {real}"
        elif not string.endswith("-"):
            string += "-"
        prev_number = real

    if not string.endswith(f", {prev_number}"):
        string += str(prev_number)

    return string
```
Could this be an alternative non-recursive way? Four more lines, but perhaps more readable?

Reply:
Agree that it is probably easier to follow. Tested performance, and they are pretty much the same (the fastest varied a bit dependent on input). Will change the variable `real` to `next_number` though, as I find it best to keep it generic in case we need a similar method for something else later. Also, this suggestion can't handle lists of length 1, but that is a simple fix of course

Reply:
Also added a test for `len(integer_list)==1`